streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -529,11 +531,13 @@ public void run() {
             cleanRun = true;
         } catch (final KafkaException e) {
             // just re-throw the exception as it should be logged already
+            // TODO: this is not true since task.update() call may throw ProcessorStateException
```

Comment:
maybe remove the comment above if it isn't correct? Seems a bit odd to leave it there